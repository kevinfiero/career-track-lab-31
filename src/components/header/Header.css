div{
  width: 100%;
  display: flex;
  justify-content: center;
}

button{
  width: 5rem;
  height: 2rem;
  margin: 1rem;
}
