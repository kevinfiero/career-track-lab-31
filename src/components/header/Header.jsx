import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { ThemeContext } from '../app/App';
import './Header.css';

const Header = ({ onClick }) => {
  const darkTheme = useContext(ThemeContext);
  const themeStyles = { backgroundColor: darkTheme ? 'white' : '#333' };

  return (  
    <div style={themeStyles}>
      <button onClick={onClick}>Toggle</button>
      <Link to="/" exact>
        <button>Home</button>
      </Link>
    </div>

  );

};

Header.propTypes = {
  onClick: PropTypes.func.isRequired
};

export default Header;
