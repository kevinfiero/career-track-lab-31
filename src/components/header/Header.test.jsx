import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import { render, cleanup } from '@testing-library/react';
import Header from './Header';

describe('Header component', () => {
  afterEach(() => cleanup());
  it('renders Header', () => {
    const { asFragment } = render(
      <Router>
        <Header/>
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
