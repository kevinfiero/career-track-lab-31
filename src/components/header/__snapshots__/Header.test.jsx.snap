// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders Header 1`] = `
<DocumentFragment>
  <a
    href="/"
  >
    <button>
      Home
    </button>
  </a>
</DocumentFragment>
`;
